### PR TITLE
Feature/location Location class methods to compute rectangles of fixed size around any position on the globe

### PR DESCRIPTION
**Pull request by ssidimoh694**
Issue #33 

### Pull Request Description
This PR introduces the `Location` utility class, which provides robust geographic calculations and normalization methods for latitude and longitude. The class includes:

- **Distance-to-angle conversion functions** for latitude and longitude, using spherical approximations.
- **rectangle generation** around a geographic center, supporting custom spans in kilometers and handling edge cases near the poles.
- **Normalization methods** for latitude and longitude to ensure values remain within valid geographic bounds.

Additionally, this PR merges comprehensive unit tests that validate:

- Correct conversion between kilometers and angular variations.
- Proper normalization of latitude and longitude values.
- Accurate polygon generation, including edge cases at the poles and around the globe.

### CODE REVIEW

File: app/src/main/java/com/github/warnastrophy/core/model/util/Location.kt [hunk 1]
```diff
@@ -1,19 +1,131 @@
...
+     * @param latitude Current latitude in degrees.
+     * @return Longitude variation in degrees.
+     */
+    fun kmToLonVariation(km: Double, latitude: Double): Double {
+      val latRad = Math.toRadians(latitude)
+      val localRadius = 2 * Math.PI * earthRadiusKm * cos(latRad)
+      if (km > localRadius) {
+        return 360.0 // full longitude range
+      }
+      return Math.toDegrees(km / earthRadiusKm)
```
Comment (Valink16) [inline app/src/main/java/com/github/warnastrophy/core/model/util/Location.kt]: Shouldn't this be km / localRadius ?

File: app/src/main/java/com/github/warnastrophy/core/model/util/Location.kt [hunk 2]
```diff
@@ -1,19 +1,131 @@
...
+     * - For each latitude, the longitudinal variation is computed via [kmToLonVariation].
+     * - Coordinates are normalized with [normalizeLat] and [normalizeLon].
+     *
+     * @param center Geographic center of the shape.
+     * @param kmLon Target longitudinal span in kilometers.
+     * @param kmLat Target latitudinal span in kilometers.
+     * @return List of normalized \[latitude, longitude\] points.
+     */
+    fun getPolygon(center: Location, kmLon: Double, kmLat: Double): List<Location> {
+      val nbrLatPoints = kotlin.math.ceil((kmLat / 2.0) / kmBetweeenPoints).toInt()
```
Comment (Valink16) [inline app/src/main/java/com/github/warnastrophy/core/model/util/Location.kt]: Smol typo on kmBetwee**e**nPoints

File: app/src/main/java/com/github/warnastrophy/core/model/util/Location.kt [hunk 3]
```diff
@@ -1,19 +1,131 @@
...
+     * @param lon Longitude in degrees.
+     * @return Longitude wrapped to \[-180.0, 180.0\].
+     */
+    fun normalizeLon(lon: Double): Double {
+      return when {
+        lon > 180 -> lon - 360
+        lon < -180 -> lon + 360
+        else -> lon
+      }
+    }
```
Comment (Valink16) [inline app/src/main/java/com/github/warnastrophy/core/model/util/Location.kt]: Would be better to use modulo division I think ? Since if the input is > 180 + 360 for example, we are still out of range

File: app/src/main/java/com/github/warnastrophy/core/model/util/utils.kt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package com.github.warnastrophy.core.model.util
+
+fun debugPrint(msg: String) {
+  val file = java.io.File("debug.txt")
+  file.appendText(msg + "\n")
+}
```
Comment (Valink16) [inline app/src/main/java/com/github/warnastrophy/core/model/util/utils.kt]: Would be better to use `Log.d` for example instead to avoid having multiple independent logs I think

File: app/src/main/java/com/github/warnastrophy/core/model/util/Location.kt [hunk 5]
```diff
@@ -1,19 +1,131 @@
...
+     * @param lat Latitude in degrees.
+     * @return Latitude clamped to \[-90.0, 90.0\].
+     */
+    fun normalizeLat(lat: Double): Double {
+      return when {
+        lat > 90.0 -> 90.0
+        lat < -90.0 -> -90.0
+        else -> lat
+      }
+    }
```
Comment (Valink16) [inline app/src/main/java/com/github/warnastrophy/core/model/util/Location.kt]: https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.ranges/coerce-in.html might be useful

File: app/src/main/java/com/github/warnastrophy/core/model/util/Location.kt [hunk 6]
```diff
@@ -1,19 +1,131 @@
...
+      if (km > localRadius) {
+        return 360.0 // full longitude range
+      }
+      return Math.toDegrees(km / earthRadiusKm)
+    }
+
+    /**
+     * Builds a set of points distributed around a center by sweeping latitudes and generating, for
+     * each, a pair of symmetric longitudes.
+     * - The number of latitudinal steps depends on `kmLat` and `kmBetweeenPoints`.
```
Comment (Valink16) [inline app/src/main/java/com/github/warnastrophy/core/model/util/Location.kt]: Smol typo on kmBetwee**e**nPoints

**Review by Valink16 — APPROVED** [review]
Good fix !
